lib/ui/painting/image_generator.h

```diff
@@ -53,6 +54,15 @@ class ImageGenerator {
   ///          always be 1 for single-frame images.
   virtual unsigned int GetFrameCount() const = 0;
 
+  /// @brief  Return the number of times to repeat, if this image is animated.
+  ///         This number does not include the first play through of each frame.
+  ///         For example, a repetition count of 4 means that each frame is
+  ///         played 5 timesand then the animation stops.
+  /// @return The number of times to repeat (not counting the first play
+  ///         through). For still (non-animated) image decoders, 0 is returned.
+  ///         For animated images that should repeat forever, -1 is returned.
+  virtual int GetRepetitionCount() const = 0;
```

Comment:
Hmm. This one is really hard to reason about and totally counter to (at least my) intuition. How about `size_t GetPlayCount() const;` instead which would be the number of time the sequence must be played. Also, playing forever, the result can be `std::limits<size_t>::max()` instead of a specific non-positive number.

Reply:
Great suggestions, thanks. I changed this up. MultiFrameCodec is still using Skia's format, but I think we could easily refactor everything else to use this clearer format in a follow-up.